drivers/gpio/gpio_shell.c

```diff
@@ -141,11 +128,53 @@ static int cmd_gpio_set(const struct shell *shell,
 }
 
 
+/* 500 msec = 1/2 sec */
+#define SLEEP_TIME_MS   500
+
+static int cmd_gpio_blink(const struct shell *shell,
```

Comment:
I do not understand this error message: drivers/gpio/gpio_shell.c:134:WARNING: Violation to rule 5.7 (Tag name should be unique) tag: shell

The shell parameter is defined like this is every shell command I have seen.

How should it be defined to pass this rule?

Reply:
@gmarull claimed this was a "false positive", so can I get some help on getting this through the process?

Reply:
@ceolin ^

Reply:
Change it to `const struct shell *sh` and the error will go away.

Reply:
I can't see how `const struct shell *shell` violates MISRA 5.7 (just read the manual and tried with https://gimpel.com/demo.html).

Reply:
MISRA is a little bit confuse about from my understanding. "A tag name shall be a unique identifier" and object is an identifier as stated in other rules, so it may be conflicting with tag name. Although https://gimpel.com/demo.html does not raise a violation, parasoft (that is the tool used by the project) does. Some of them can be seen here https://codescan.zephyrproject.org:8443/grs/dtp/explorers/violations?buildId=v2.7.0-rc1-48-g6e26512bd5&filterId=2&rule=MISRAC2012-RULE_5_7-a

```
struct multiboot_info multiboot_info;
...
time_t timeutil_timegm(const struct tm *tm)
```

I think the main problem is that MISRA-C does not list this case in this rule examples.

Reply:
Fixed!

Reply:
@gmarull ^

Reply:
Thanks for the clarification. MISRA is clearly vague here, to the point some checkers warn and others don't... :-)